### PR TITLE
Adds version endpoint and service

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm/issues/1707

This PR extracts the version endpoint and service. Later I think we can add healthz but we'll need an interface for the health checks. This is for use in `azure-operator`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,11 @@
+[![CircleCI](https://circleci.com/gh/giantswarm/microendpoint.svg?style=svg)](https://circleci.com/gh/giantswarm/microendpoint)
```
Comment (rossf7) [inline README.md]: Basic readme

File: endpoint/version/endpoint.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
+package version
```
Comment (rossf7) [inline endpoint/version/endpoint.go]: This is the extracted endpoint.

File: service/version/service.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package version
```
Comment (rossf7) [inline service/version/service.go]: I also extracted the service since its always the same.

File: glide.lock [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  version: a9ca6725cbbea455e61c6bc8a1ed28e81eb3493b
+  subpackages:
+  - endpoint
+  - log
+  - transport/http
+- name: github.com/go-logfmt/logfmt
+  version: 390ab7935ee28ec6b286364bba9b4dd6410cb3d5
+- name: github.com/go-stack/stack
+  version: 100eb0c0a9c5b306ca2fb4f165df21d80ada4b82
+- name: github.com/juju/errgo
```
Comment (JosephSalisbury) [inline glide.lock]: I can't see where this is used?

File: service/version/service.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		return nil, microerror.Maskf(invalidConfigError, "description commit must not be empty")
+	}
+	if config.GitCommit == "" {
+		return nil, microerror.Maskf(invalidConfigError, "git commit must not be empty")
+	}
+	if config.Name == "" {
+		return nil, microerror.Maskf(invalidConfigError, "name must not be empty")
+	}
+	if config.Source == "" {
+		return nil, microerror.Maskf(invalidConfigError, "name must not be empty")
```
Comment (JosephSalisbury) [inline service/version/service.go]: source

File: glide.lock (hunk 4 above)
Comment (rossf7) [inline glide.lock]: Fixed

File: service/version/service.go (hunk 5 above)
Comment (rossf7) [inline service/version/service.go]: fixed

File: glide.lock (hunk 4 above)
Comment (rossf7) [inline glide.lock]: Both imports are from `micrologger` it's not using `microerror` yet.

**Comment by rossf7** [comment]
@JosephSalisbury PTAL. Error message bug is fixed and the imports are from `micrologger`. I think it should use `microerror` but can we do that later?

File: endpoint/version/endpoint.go (hunk 2 above)
Comment (kopiczko) [inline endpoint/version/endpoint.go]: I like this is also state of the art template for new endpoints

File: glide.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package: github.com/giantswarm/microendpoint
+import:
+- package: github.com/giantswarm/microerror
+  version: master
+- package: github.com/giantswarm/micrologger
+  version: master
```
Comment (kopiczko) [inline glide.yaml]: I don't see explicit go-kit dep

**Comment by rossf7** [comment]
@kopiczko the go-kit dep is now set.

File: endpoint/version/response.go [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+func DefaultResponse() *Response {
+	return &Response{
+		Description: "",
+		GitCommit:   "",
+		GoVersion:   "",
+		Name:        "",
+		OSArch:      "",
+		Source:      "",
+	}
+}
```
Comment (kopiczko) [inline endpoint/version/response.go]: Something wrong with your settings :D

File: glide.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+- package: github.com/go-kit/kit
+  version: v0.5.0
+  subpackages:
+  - log
+  - transport/http
```
Comment (kopiczko) [inline glide.yaml]: Did you add subpackages by hand?

Comment (kopiczko) [inline glide.yaml]: BTW I don't think you need to specify log package. It's specified by micrologger.

File: endpoint/version/response.go (hunk 7 above)
Comment (rossf7) [inline endpoint/version/response.go]: This is Code again :( You have to add an extension for the newline endings. It's working so not sure how this got through. I'll look out for it.

Comment (kopiczko) [inline endpoint/version/response.go]: Not a big deal. Don't spend to much time. I won't say a word again <3

**Comment by JosephSalisbury** [comment]
IT'S THE TRIPLE WHAMMYYYYYYYY

**Comment by rossf7** [comment]
All green. Thanks! <3

**Pull request merged**